pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -831,4 +832,24 @@ public static boolean like(String inputStr, String likePatternStr) {
     String regexPatternStr = RegexpPatternConverterUtils.likeToRegexpLike(likePatternStr);
     return regexpLike(inputStr, regexPatternStr);
   }
+
+  /**
+   * Checks whether the input string can be parsed into a json node or not. Useful for scenarios where we want
+   * to filter out malformed json. In case of nulls, it is treated as valid json as in partial-upsert we might
+   * want to treat that column value as valid.
+   *
+   * @param inputStr Input string to test for valid json
+   */
+  @ScalarFunction
+  public static boolean isJson(String inputStr) {
+    try {
+      if (inputStr == null) {
```

Comment:
In order to take `null`, you need to annotate the function with `nullableParameters`. Please also annotate the parameter as `@Nullable`. You might also want to add alias `is_json`

Reply:
updated! Thanks!

Reply:
nit: Semantically speaking, `null` in databases means "unknown" or "missing", and hence one might expect `isJson` to return false for null.

To address this, we could consider adding a two argument variant of this

```
public static boolean isJson(String inputStr, boolean acceptNull) {
  try {
    if (inputStr == null) {
      return acceptNull;
    }
   ...
  }
}
```

Reply:
Yeah sounds good to me! Updated accordingly!

Reply:
The current semantic is quite confusing. I feel we should just not take `null` and the framework will simply return `null` when the input is `null`
```suggestion
  @ScalarFunction(names = {"isJson", "is_json"})
  public static boolean isJson(String inputStr) {
    try {
      JsonUtils.stringToJsonNode(inputStr);
      return true;
    } catch (Exception e) {
      return false;
    }
  }
```

cc @ankitsultana

Reply:
@Jackie-Jiang the suggestion will return `false` for `null` right as it would be an exception? Do you mean to return Object instead?

Reply:
Something like this?
```
@ScalarFunction(nullableParameters = true, names = {"isJson", "is_json"})
  public static Object isJson(@Nullable String inputStr) {
    try {
      if (inputStr == null) {
        return null;
      }
      JsonUtils.stringToJsonNode(inputStr);
      return true;
    } catch (Exception e) {
      return false;
    }
  }
```

Reply:
If there is no `nullableParameters = true` in the annotation, it will return `null` whenever the argument is `null`. No need to explicitly return `null`

Reply:
Ah that makes it much simpler then! We can just use `isJson(str) == false` to find invalid jsons. Let me update accordingly!